lib/rmt/lockfile.rb

```diff
@@ -4,13 +4,13 @@ class RMT::Lockfile
 
   class << self
     def create_file
-      if File.exist?(LOCKFILE_LOCATION)
-        raise RMT::Lockfile::ExecutionLockedError
-      else
-        dirname = File.dirname(LOCKFILE_LOCATION)
-        FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
-        File.open(LOCKFILE_LOCATION, 'w') { |f| f.write(Process.pid) }
-      end
+      dirname = File.dirname(LOCKFILE_LOCATION)
+      FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
+
+      f = File.open(RMT::Lockfile::LOCKFILE_LOCATION, File::RDWR | File::CREAT)
+      Timeout.timeout(0.001) { f.flock(File::LOCK_EX) }
```

Comment:
Timeout is not needed. See the previous PR comments -- use `File::LOCK_EX | File::LOCK_NB` mode.
From the docu:
```
Returns false if File::LOCK_NB is specified and the operation would otherwise have blocked.
```